internal/vds/vds.cpp

```diff
@@ -427,10 +427,32 @@ struct response fetch_horizon(
                  k[2] = std::round( k[2] + 1) - 1;
             }
 
-            samples[i][  iline.dimension() ] = ij[0] + 0.5;
-            samples[i][  xline.dimension() ] = ij[1] + 0.5;
-            samples[i][ sample.dimension() ] =  k[2] + 0.5;
-            ++i;
+            ij[0] += 0.5;
+            ij[1] += 0.5;
+             k[2] += 0.5;
+
+            if (not inrange(iline, ij[0]) or not inrange(xline, ij[1])) {
```

Comment:
Because those bounds check a voxel in 3 dimensions. And we want different behaviour based on whether it's out of bounds in the vertical domain or the horizontal